File: Python/Program/Antivirus.py
from Software import Software

class Antivirus(Software):
    def __init__(self, tipe_proteksi="", tanggal_expired="", nama="", versi="", developer=""):
        super().__init__(nama, versi, developer)
        self.__tipe_proteksi = tipe_proteksi
        self.__tanggal_expired = tanggal_expired
    
    def set_tipe_proteksi(self, tipe_proteksi):
        self.__tipe_proteksi = tipe_proteksi
    
    def set_tanggal_expired(self, tanggal_expired):
        self.__tanggal_expired = tanggal_expired
    
    def get_tipe_proteksi(self):
        return self.__tipe_proteksi
    
    def get_tanggal_expired(self):
        return self.__tanggal_expired

File: CPP/Program/Antivirus.cpp
#ifndef ANTIVIRUS_CPP
#define ANTIVIRUS_CPP

#include <iostream>
#include <string>
#include "Software.cpp"
using namespace std;

class Antivirus : public Software {
private:
    string tipeProteksi; // Real-time, On-demand, dll
    string tanggalExpired;
    
public:
    Antivirus() {
    }
    
    Antivirus(string tipeProteksi, string tanggalExpired, string nama, string versi, string developer) : Software(nama, versi, developer) {
        this->tipeProteksi = tipeProteksi;
        this->tanggalExpired = tanggalExpired;
    }
    
    void setTipeProteksi(string tipeProteksi) {
        this->tipeProteksi = tipeProteksi;
    }
    
    void setTanggalExpired(string tanggalExpired) {
        this->tanggalExpired = tanggalExpired;
    }
    
    string getTipeProteksi() {
        return this->tipeProteksi;
    }
    
    string getTanggalExpired() {
        return this->tanggalExpired;
    }
    
    ~Antivirus() {
    }
};

#endif // ANTIVIRUS_CPP